### ISSUE TITLE
Some minor UI flaws

### ISSUE THREAD

**Issue by ninsbl** [rashadkm/webgrass#20]
Hi,

now, without authentication, I discovered some minor issues in the GUI:
1) locations and mapsets do not order (alphabeticaly)
2) After a location is chosen, the location column gets very long and one has to scoll down to the start WebGRASS button
3) Typo in "Add Raster" button (there it says "Add rater")
4) Menu is shown vertical instead of horizontal.
5) Modules have no own window and options are spread around the UI.

see attached screenshots:

![webgrass](https://cloud.githubusercontent.com/assets/4246620/17766759/f8f4e706-652c-11e6-8aef-6d1ff6292fb9.JPG)
![webgrass_location_mapset](https://cloud.githubusercontent.com/assets/4246620/17766758/f8ee7c2c-652c-11e6-821b-58d5183d3284.JPG)
![webgrass_modules](https://cloud.githubusercontent.com/assets/4246620/17766961/1a294a24-652e-11e6-99d6-cca8eea147e7.JPG)

But is otherwise looks quite cool already!

**Comment by mayank33** [comment]
This happens in some of the people. (with me also). Try this resources folder. https://drive.google.com/open?id=0ByvUEyWRSot9UXN5b2s0NER5NHM

Will give a fix for it once many people have given the feedback. Replace this resources folder with the old one and then compile. It will definitely work. Screen shots should look like this https://trac.osgeo.org/grass/wiki/GSoC/2016/WebGrass (wiki page)

Most of the issues will be solved

**Comment by mayank33** [comment]
Did this work??

**Comment by ninsbl** [comment]
Yes, now, with the new resource folder, the Main UI looks like an application for me too.
Points 4 and 5 from above seem to be solved!

**Comment by mayank33** [comment]
Yeah will fix those first 3 points.

**Comment by mayank33** [comment]
Please check. All issues are taken care of.

**Comment by ninsbl** [comment]
Hi, 
1  and 3 are fixed. Here, locations and mapsets order all entries starting with uppercase first and then all starting with lower case.
However 2, 4, and 5 are still present in a new, fresh and clean clone from github.

I noticed that I now also had to manually modify:
.webgrass/rc
(in addition to ../scripts/init_grass.sh and compilation options in cmake (here I actually use ccmake .. instead of cmake .. as described in the readme)).

Would be nice if the README or README.md could be updated, so people get pointed to those required changes for installation. I can provide you with input if you like ...

**Comment by ninsbl** [comment]
BTW. it seems that ./webgrass/rc overrides the users choice of location and mapset.

**Comment by mayank33** [comment]
rc file is written when selection is made (you dont have to modify it manually). try with the new resources (I mentioned it earlier also). if not please give a screenshot of what happening. with the new resources, 4 and 5 will be solved

https://drive.google.com/file/d/0ByvUEyWRSot9UXN5b2s0NER5NHM/view

**Comment by mayank33** [comment]
you should get the screenshot like this https://trac.osgeo.org/grass/attachment/wiki/GSoC/2016/WebGrass/ModuleUI1.png

**Comment by ninsbl** [comment]
Now I removed my old local copy, cloned this repo and applied #24 locally.
Currently, in order to make WebGRASS start properly I have to manually edit:
- scripts/init_grass.sh (e.g. adjust paths to my GRASS installation, which is in /usr/local and not /usr/lib. I made exactly the same changes before in the ccmake gui, so they do not seem to be passed to scripts/init_grass.sh
- .webgrass/rc (It seems that this rc file is used at the moment regardless user choices... Before I edited the rc file I got an error message on the console, saying something like: "Could not find /home/mayank/grassdata")

Adding the resources from your dropbox manually did solve the issues 4 + 5 earlier, so I dd not test it again. However, it would be great if the correct resources would be shipped with this repo and would not have to be added manually after cloning the repo...

In order to have things reproducible, it would be beneficial, if all required installation steps were documented in the readme...

**Comment by mayank33** [comment]
Readme is updated. The data which you want to use should be in folder "grassdata" in your home. It automatically take this and look like this https://trac.osgeo.org/grass/attachment/wiki/GSoC/2016/WebGrass/Startup.png

Here you can see GRASS DATA DIR points to your data. Please check if it is taking correct folder or not.

We made it like it will take "grassdata" folder from your home.

When you make selection, ".webgrass/rc" is updated and is used. It will work properly if grassdata is present in your home.

This thing is updated in the readme.  

Please paste the screenshot of the first page if problem still exist.

**Comment by ninsbl** [comment]
The issue with the location and mapset dialogue is still the same. Anyway, I added a new screenshot which illustrates the problem a bit better. It seems that the dialogue boxes are extended to the length of the list of locations or mapsets (depending on what is longer). See:
![webgrass_location_mapset_2](https://cloud.githubusercontent.com/assets/4246620/18128299/53ed7a08-6f85-11e6-8984-d56df119f45b.JPG)
After resizing the browser window also the mapset selection box get resized to show all mapsets instead of being scrollable...

Regarding the resource folder: I do not quite understand why this is not integrated in the github repo, but seems to remain on a personal Dropbox folder which has to be added manually after cloning the repo? Does the current configuration work on some machines but not on other?

For the handling of grassdata folder, init_grass.sh... I will open a new issue...

**Comment by mayank33** [comment]
Yes, for the resource folder, the issue is the same. Resources on dropbox works for me but for my mentors, the resources on the repo works. That's why we are testing to know which one works better for the people. I will check this(scrollable) issue once again. It was working for me.
